src/CoreWCF.Primitives/src/CoreWCF/Channels/ByteStreamMessageEncoder.cs

```diff
@@ -0,0 +1,262 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using System.Xml;
+using CoreWCF.Runtime;
+using CoreWCF.Runtime.Diagnostics;
+
+namespace CoreWCF.Channels
+{
+    internal class ByteStreamMessageEncoder : MessageEncoder, ITraceSourceStringProvider, IWebMessageEncoderHelper, IStreamedMessageEncoder
+    {
+        private string _traceSourceString;
+        private readonly string _maxSentMessageSizeExceededResourceString;
+        private readonly XmlDictionaryReaderQuotas _quotas;
+        private readonly XmlDictionaryReaderQuotas _bufferedReadReaderQuotas;
+
+        /// <summary>
+        /// Specifies if this encoder produces Messages that provide a body reader (with the Message.GetReaderAtBodyContents() method) positioned on content.
+        /// See the comments on 'IWebMessageEncoderHelper' for more info.
+        /// </summary>
+        private bool _moveBodyReaderToContent = false; // false because we want the ByteStreamMessageEncoder to be compatible with previous releases
+
```

Comment:
The 4.0 compat functionality behind this bool can be removed

Reply:
This is addressed in b0c21a3804140f47b064720903835a1427a24357.